conformance/tests/gatewayclass-supported-version-condition.go

```diff
@@ -0,0 +1,102 @@
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/consts"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, GatewayClassSupportedVersionCondition)
+}
+
+var GatewayClassSupportedVersionCondition = suite.ConformanceTest{
+	ShortName: "GatewayClassSupportedVersionCondition",
+	Features: []features.FeatureName{
+		features.SupportGateway,
+	},
+	Description: "A GatewayClass should set the SupportedVersion condition based on the presence and version of Gateway API CRDs in the cluster",
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+		gwc := types.NamespacedName{Name: s.GatewayClassName}
+
+		t.Run("SupportedVersion condition should be set correctly", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), s.TimeoutConfig.DefaultTestTimeout)
+			defer cancel()
+
+			// Ensure GatewayClass conditions are as expected before proceeding
+			kubernetes.GWCMustHaveAcceptedConditionTrue(t, s.Client, s.TimeoutConfig, gwc.Name)
+			kubernetes.GWCMustHaveSupportedVersionConditionAny(t, s.Client, s.TimeoutConfig, gwc.Name)
+
+			// Retrieve the GatewayClass CRD
+			crd := &apiextv1.CustomResourceDefinition{}
+			crdName := types.NamespacedName{Name: "gatewayclasses.gateway.networking.k8s.io"}
+			err := s.Client.Get(ctx, crdName, crd)
+			require.NoErrorf(t, err, "error getting GatewayClass CRD: %v", err)
+
+			if crd.Annotations != nil {
+				// Remove the bundle version annotation if it exists
+				if _, exists := crd.Annotations[consts.BundleVersionAnnotation]; !exists {
+					t.Fatalf("Annotation %q does not exist on CRD %s", consts.BundleVersionAnnotation, crdName)
+				}
+				delete(crd.Annotations, consts.BundleVersionAnnotation)
+				if err := s.Client.Update(ctx, crd); err != nil {
+					t.Fatalf("Failed to update CRD %s: %v", crdName, err)
+				}
+			}
+
+			// Ensure the SupportedVersion status condition is false
+			kubernetes.GWCMustHaveSupportedVersionConditionFalse(t, s.Client, s.TimeoutConfig, gwc.Name)
+
+			// Add the bundle version annotation
+			crd.Annotations[consts.BundleVersionAnnotation] = consts.BundleVersion
+			if err := s.Client.Update(ctx, crd); err != nil {
+				t.Fatalf("Failed to update CRD %s: %v", crdName, err)
+			}
+
+			// Ensure the SupportedVersion status condition is true
+			kubernetes.GWCMustHaveSupportedVersionConditionTrue(t, s.Client, s.TimeoutConfig, gwc.Name)
+
+			// Set the bundle version annotation to an unsupported version
+			crd.Annotations[consts.BundleVersionAnnotation] = "v0.0.0"
+			if err := s.Client.Update(ctx, crd); err != nil {
+				t.Fatalf("Failed to update CRD %s: %v", crdName, err)
+			}
+
+			// Ensure the SupportedVersion is false
+			kubernetes.GWCMustHaveSupportedVersionConditionFalse(t, s.Client, s.TimeoutConfig, gwc.Name)
```

Comment:
Some implementations like GKE will ~immediately overwrite this CRD with the original version, can you add that as an acceptable outcome of this test?